### PR TITLE
Add ImageOutput to egress EncodedOutput utils

### PR DESCRIPTION
**Pull request by biglittlebigben**

### CODE REVIEW

File: egress/types.go [hunk 1]
```diff
@@ -35,12 +36,20 @@ type EncodedOutput interface {
 	GetFileOutputs() []*livekit.EncodedFileOutput
 	GetStreamOutputs() []*livekit.StreamOutput
 	GetSegmentOutputs() []*livekit.SegmentedFileOutput
+	ImageOutput
 }
 
 type EncodedOutputDeprecated interface {
```
Comment (frostbyte73) [inline egress/types.go:42]: This is for deprecated requests which don't implement `Get_Outputs()`

File: egress/types.go [hunk 2]
```diff
@@ -35,12 +36,20 @@ type EncodedOutput interface {
...
 	GetFile() *livekit.EncodedFileOutput
 	GetStream() *livekit.StreamOutput
 	GetSegments() *livekit.SegmentedFileOutput
+	GetFileOutputs() []*livekit.EncodedFileOutput
+	GetStreamOutputs() []*livekit.StreamOutput
+	GetSegmentOutputs() []*livekit.SegmentedFileOutput
+}
+
+type ImageOutput interface {
+	GetImageOutputs() []*livekit.ImageOutput
```
Comment (frostbyte73) [inline egress/types.go:52]: Is there a reason for this to be separate?

Comment (frostbyte73) [reply egress/types.go:52]: nvm, we'll now have 2 kinds of deprecated requests - the oldest having only GetX, one having GetXOutputs without Images, and the newest having GetXOutputs including images

**Comment by biglittlebigben** [comment]
The answer to both remarks is that I was hoping to replace the types in https://github.com/livekit/egress/blob/2b48c8a5b576017cdfe71944dcb9713a5ca17eb8/pkg/config/output.go#L43 with the ones from protocol above to avoid type duplication.

That being said, after checking, the EncodedOutputDeprecated type in egress never uses the Get_Outputs getters, or the ImageOutput interface directly, so we can clean this up indeed.

File: egress/types.go (hunk 1 above)
Comment (davidzhao) [reply egress/types.go:42]: I think it's ok to leave it out of the legacy API? i.e. if you want new features then use the new API.

**Review by frostbyte73 — APPROVED** [review]

**Pull request merged**